### PR TITLE
deadbeef: update to 1.9.6

### PR DESCRIPTION
**Pull request by zlice**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, **x86_64**
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - **x86_64-musl**

#### Comments

Runs, plays music

### CODE REVIEW

**Comment by zlice** [comment]
for `libispatch/devel` i assume this is true? `=> WARNING: usr/share/man/man3 should be in -devel package`

**Comment by classabbyamp** [comment]
correct, man section 3 is library/api documentation so it should go in -devel

**Comment by zlice** [comment]
aarch64 seems to crossbuild for me locally

**Comment by classabbyamp** [comment]
why is clang needed in general?

File: srcpkgs/libdispatch/template [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Template file for 'libdispatch'
+pkgname=libdispatch
+version=5.10.1
+revision=1
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=MinSizeRel"
```
[the remark this answers is not in the document]
Comment (zlice) [reply srcpkgs/libdispatch/template:6]: i was just copying from alpine to see if i missed something that may change the arm cross builds  but i guess not.

**Comment by zlice** [comment]
> why is clang needed in general?

https://github.com/DeaDBeeF-Player/deadbeef?tab=readme-ov-file#building-deadbeef-from-source

That is what they use to build. I don't think gcc builds.

File: srcpkgs/deadbeef/template [hunk 2]
```diff
@@ -1,37 +1,39 @@
 # Template file for 'deadbeef'
 pkgname=deadbeef
-version=1.8.8
-revision=3
+version=1.9.6
+revision=1
 create_wrksrc=yes
```
Comment (classabbyamp) [inline srcpkgs/deadbeef/template:5]:
```suggestion
```

File: srcpkgs/deadbeef/template [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
+ https://github.com/DeaDBeeF-Player/ddb_output_pw/archive/${_output_pw_commit}.tar.gz>output_pw-${_output_pw_commit}.tar.gz
  https://github.com/DeaDBeeF-Player/mp4p/archive/${_mp4p_commit}.tar.gz>mp4p-${_mp4p_commit}.tar.gz"
-checksum="a667d55b787d76bc9d3593882b57ae9a449a99e93e1c11f85ec638566e10877d
- c5eeb3fa116d51d251fe7c7e5c44798680f6478c43a1fffdcb28c4caf1c68286
- d0ba1b4ad062cb5c056c4a59ed55769d79371fc9b3bf9aaa0ca48f508e3eba75"
-LDFLAGS+=" -Wl,-z,stack-size=1048576"
+checksum="53ed535335e637437adf77eb4bcadb781eddbf5539f74373d5a863389df5e2d0
+444d4d89edbd51b9d2305c83a49e18949e0f21a42eec2a95ce03efd752a81049
+59115ddcd0378aa2f5914138c5c256198d66339bfbb3d65389b9bf4fa327f9ee
+3b5bdbcb2808d12b9f5af630a91e77be1036aeb487d5fa0a323ce8080918439b"
```
Comment (classabbyamp) [inline srcpkgs/deadbeef/template:31]:
```suggestion
checksum="53ed535335e637437adf77eb4bcadb781eddbf5539f74373d5a863389df5e2d0
 444d4d89edbd51b9d2305c83a49e18949e0f21a42eec2a95ce03efd752a81049
 59115ddcd0378aa2f5914138c5c256198d66339bfbb3d65389b9bf4fa327f9ee
 3b5bdbcb2808d12b9f5af630a91e77be1036aeb487d5fa0a323ce8080918439b"
skip_extraction="libretro-${_libretro_commit}.tar.gz
 output_pw-${_output_pw_commit}.tar.gz
 mp4p-${_mp4p_commit}.tar.gz"
```

File: srcpkgs/deadbeef/template [hunk 4]
```diff
@@ -40,14 +42,20 @@ fi
...
 post_extract() {
 	mv deadbeef-${version}/* .
-	rm -rf external/apbuild
 	rm -rf external/mp4p
-	mv -v apbuild-${_apbuild_commit} external/apbuild
+	rm -rf external/ddb_dsp_libretro
+	rm -rf external/ddb_output_pw
+	mv -v ddb_dsp_libretro-${_libretro_commit} external/ddb_dsp_libretro
+	mv -v ddb_output_pw-${_output_pw_commit} external/ddb_output_pw
 	mv -v mp4p-${_mp4p_commit} external/mp4p
```
Comment (classabbyamp) [inline srcpkgs/deadbeef/template:50]:
```
vsrcextract -C external/ddb_dsp_libretro libretro-${_libretro_commit}.tar.gz
vsrcextract -C external/ddb_output_pw output_pw-${_output_pw_commit}.tar.gz
vsrcextract -C external/mp4p p4p-${_mp4p_commit}.tar.gz
```

File: srcpkgs/deadbeef/template (hunk 2 above)
Comment (zlice) [reply srcpkgs/deadbeef/template:5]: think this is still needed

```
=> deadbeef-1.9.6_1: running post_extract ...
mv: cannot stat 'deadbeef-1.9.6/*': No such file or directory
=> ERROR: deadbeef-1.9.6_1: post_extract: 'mv deadbeef-${version}/* .' exited with 1
=> ERROR:   in post_extract() at srcpkgs/deadbeef/template:46
```

File: srcpkgs/deadbeef/template (hunk 3 above)
Comment (zlice) [reply srcpkgs/deadbeef/template:31]: not sure this works with the vsrcextract?

```
=> deadbeef-1.9.6_1: running post_extract ...
bsdtar: Error opening archive: Failed to open '/host/sources/deadbeef-1.9.6/p4p-156195ccb635f016dc34b89425bfbecf046c90d4.tar.gz'
=> ERROR: deadbeef-1.9.6_1: post_extract: '$TAR_CMD ${sc:+"$sc"} ${dst:+-C "$dst"} -x --no-same-permissions --no-same-owner -f $archive' exited with 1
=> ERROR:   in vextract() at common/environment/setup/archive.sh:69
=> ERROR:   in vsrcextract() at common/environment/setup/archive.sh:163
=> ERROR:   in post_extract() at srcpkgs/deadbeef/template:53
```

File: srcpkgs/deadbeef/template (hunk 2 above)
Comment (classabbyamp) [reply srcpkgs/deadbeef/template:5]: not if you apply the other comments (skip extraction and vsrcextract)

File: srcpkgs/deadbeef/template (hunk 3 above)
Comment (zlice) [reply srcpkgs/deadbeef/template:31]: ah ya got the `mv`* out too, error below about permissions when extracting is still there. not familiar with the vsrcextract

Comment (zlice) [reply srcpkgs/deadbeef/template:31]: guessing since extraction is skipped that file isn't in `/host/sources` ? there's nothing under `/host`

Comment (classabbyamp) [reply srcpkgs/deadbeef/template:31]: no, it should be there. try cleaning and rebuilding

File: srcpkgs/deadbeef/template [hunk 5]
```diff
@@ -1,53 +1,65 @@
...
-	rm -rf external/apbuild
+#	mv deadbeef-${version}/* .
 	rm -rf external/mp4p
-	mv -v apbuild-${_apbuild_commit} external/apbuild
-	mv -v mp4p-${_mp4p_commit} external/mp4p
+	rm -rf external/ddb_dsp_libretro
+	rm -rf external/ddb_output_pw
+	vsrcextract -C external/ddb_dsp_libretro libretro-${_libretro_commit}.tar.gz
+	vsrcextract -C external/ddb_output_pw output_pw-${_output_pw_commit}.tar.gz
+	vsrcextract -C external/mp4p p4p-${_mp4p_commit}.tar.gz
```
Comment (classabbyamp) [inline srcpkgs/deadbeef/template:54]:
```suggestion
	vsrcextract -C external/mp4p mp4p-${_mp4p_commit}.tar.gz
```

Comment (zlice) [reply srcpkgs/deadbeef/template:54]: that **m** ah. k think most of these are covered

**Comment by zlice** [comment]
it looks like configure, by default, is getting `--host` twice?

https://github.com/void-linux/void-packages/blame/master/common/environment/configure/gnu-configure-args.sh#L15

looks like line 15 and 43 will set twice? why is that?

**config.log**

```
  $ ./configure --prefix=/usr --sysconfdir=/etc --sbindir=/usr/bin --bindir=/usr/bin --mandir=/usr/share/man --infodir=/usr/share/info --localstatedir=/var --host=x86_64-unknown-linux-gnu --build=x86_64-unknown-linux-gnu '--libdir=${exec_prefix}/lib64' --host=aarch64-linux-gnu --with-sysroot=/usr/aarch64-linux-gnu --with-libtool-sysroot=/usr/aarch64-linux-gnu --disable-oss --disable-lfm --disable-gtk2 --disable-libretro
```

**Comment by zlice** [comment]
i feel like this is hacky and shouldn't have to be done. deadbeef may have changed behavior, but since kodi is having similar cross compile issues i'm not so sure.

**Comment by classabbyamp** [comment]
what a mess lol

**Pull request merged**